plugins/license-gather-plugin/src/test/kotlin/com/github/vlsi/gradle/license/VerifyLicenseCompatibilityTaskTest.kt

```diff
@@ -0,0 +1,215 @@
+        }
+    }
+
+    private val gradleRunner = GradleRunner.create().withPluginClasspath()
+
+    @TempDir
+    lateinit var projectDir: Path
+
+    fun Path.write(text: String) = this.toFile().writeText(text)
+
+    @ParameterizedTest
+    @MethodSource("gradleVersionAndSettings")
+    fun `licenseGathering works`(gradleVersion: String, extraSettings: String) {
+        projectDir.resolve("settings.gradle").write(
+            """
+            rootProject.name = 'sample'
+            $extraSettings
+        """
+        )
+        projectDir.resolve("build.gradle").write(
+            /* language=groovy */ """
+            import com.github.vlsi.gradle.license.GatherLicenseTask
+            import com.github.vlsi.gradle.license.VerifyLicenseCompatibilityTask
+            import com.github.vlsi.gradle.release.AsfLicenseCategory
+            import com.github.vlsi.gradle.license.api.LicenseExpressionExtensions
+            import com.github.vlsi.gradle.license.api.SpdxLicense
+            import com.github.vlsi.gradle.license.api.SimpleLicense
+
+            plugins {
+                id('java')
+                id('com.github.vlsi.license-gather')
+            }
+            group = 'org.example'
+            version = '0.0.1'
+
+            repositories {
+                mavenCentral()
+                jcenter()
+            }
+
+            dependencies {
+                runtimeOnly("org.slf4j:slf4j-api:1.7.25")
+                runtimeOnly("org.junit.jupiter:junit-jupiter:5.4.2")
+                runtimeOnly("org.jodd:jodd-core:5.0.6")
+                runtimeOnly("org.jetbrains.lets-plot:lets-plot-batik:2.1.0")
+                runtimeOnly("org.jetbrains.lets-plot:lets-plot-kotlin-jvm:3.0.2")
+            }
+
+            def generateLicense = tasks.register("generateLicense", GatherLicenseTask.class) {
+                configurations.add(project.configurations.runtimeClasspath)
+                ignoreMissingLicenseFor(SpdxLicense.BSD_2_Clause)
+                ignoreMissingLicenseFor(SpdxLicense.MIT)
+            }
+
+            tasks.register("verifyLicenses", VerifyLicenseCompatibilityTask.class) {
+                metadata.from(generateLicense)
+                allow(SpdxLicense.EPL_2_0) {
+                    because("JUnit is OK")
+                }
+                allow(new SimpleLicense("The W3C License", uri("http://www.w3.org/TR/2004/REC-DOM-Level-3-Core-20040407/java-binding.zip"))) {
+                    because("ISSUE-42: John Smith decided the license is OK")
+                }
+                // No reason, for test purposes
+                allow(LicenseExpressionExtensions.orLater(SpdxLicense.SAX_PD))
+                // https://touk.pl/blog/2018/05/28/testing-kotlin-with-spock-part-2-enum-with-instance-method/
+                AsfLicenseCategory.@A.licenseExpressions.forEach {
+                    allow(it) {
+                        because("The ASF category A is allowed")
+                    }
+                }
```

Comment:
This overload for `allow` would make things a bit more ergonomic to write. WDYT?

```suggestion
                allow(AsfLicenseCategory.@A) {
                    because("The ASF category A is allowed")
                }
```

Reply:
What I do not like is 6 overloads for allow-reject-unknown and license-licenseExpresion.

Adding category adds 3 more, and it probaby makea sense to add LicenseCategory interface to avoid adding more and more overloads

Reply:
I made `License: LicenseExpressionSet` and `LicenseExpression: LicenseExpressionSet`, and `AsfLicenseCategory: LicenseExpressionSet`